src/model.jl

```diff
@@ -258,18 +333,40 @@ function _build_model(m::JuMP.Model; case::Case, storage::Storage,
     if load_shed_enabled
         injections = JuMP.@expression(m, injections + load_bus_map * load_shed)
     end
-    withdrawls = JuMP.@expression(m, bus_demand)
+    withdrawals = JuMP.@expression(m, bus_demand)
     if storage_enabled
         injections = JuMP.@expression(m, injections + storage_map * storage_dis)
-        withdrawls = JuMP.@expression(m, withdrawls + storage_map * storage_chg)
+        withdrawals = JuMP.@expression(m, withdrawals + storage_map * storage_chg)
     end
-    JuMP.@constraint(m, powerbalance, (injections .== withdrawls))
+    if demand_flexibility.enabled
+        injections = JuMP.@expression(m, injections + load_bus_map * load_shift_dn)
+        withdrawals = JuMP.@expression(m, withdrawals + load_bus_map * load_shift_up)
+    end
+    JuMP.@constraint(m, powerbalance, (injections .== withdrawals))
     println("powerbalance, setting names: ", Dates.now())
     for i in sets.bus_idx, j in hour_idx
         JuMP.set_name(powerbalance[i, j],
                       "powerbalance[" * string(i) * "," * string(j) * "]")
     end
 
+    if load_shed_enabled
+        if demand_flexibility.enabled
+            JuMP.@constraint(
+                m, 
+                load_shed_ub[i in 1:sets.num_load_bus, j in 1:interval_length], 
+                load_shed[i, j] <= bus_demand[sets.load_bus_idx[i], j] 
+                    + load_shift_up[i, j]
+                    - load_shift_dn[i, j]
+            )
+        else
+            JuMP.@constraint(
+                m, 
+                load_shed_ub[i in 1:sets.num_load_bus, j in 1:interval_length], 
+                load_shed[i, j] <= bus_demand[sets.load_bus_idx[i], j]
+            )
+        end
+    end
+
```

Comment:
Can we build up the right-hand-side of this constraint (based on the ordering of variables/parameters here, not based on the 'traditional' RHS definition which is parameters only) using a JuMP expression? I.e. start with the bus demand, add load shifting as necessary, and then constrain based on this expression?

Or equivalently, start with an expression containing load shedding, add load shifting as necessary, and constrain this expression by the bus demand?

Reply:
Yeah, that's a good idea. That would definitely be cleaner.

Reply:
I believe I have implemented something similar to what you were describing.